File: JobMatchingSystem.FE/src/models/job.ts
import { Taxonomy } from './taxonomy';

// Định nghĩa các loại công việc
export type JobType = 'FullTime' | 'PartTime' | 'Remote' | 'Other';
export type JobStatus = 'Draft' | 'Rejected' | 'Moderated' | 'Opened' | 'Closed';
export const CandidateJobStatus = { Pending: 0, RejectCv: 1, Processing: 2, Fail: 3, Pass: 4, } as const;
export type CandidateJobStatus = typeof CandidateJobStatus[keyof typeof CandidateJobStatus];

export interface Job {
  jobId: number;
  title: string;
  description: string;
  requirements: string;
  benefits: string;
  salaryMin?: number;
  salaryMax?: number;
  location: string;
  experienceYear?: number;
  jobType: JobType;
  status: JobStatus;
  viewsCount: number;
  companyId: number;
  recuiterId: number;
  verifiedBy?: number;
  createdAt: string;
  openedAt?: string;
  expiredAt?: string;
  isDeleted: boolean;
  taxonomies: Taxonomy[];
  isReport?: boolean;
  isApply?: boolean;
  isSave?: boolean;
  applyCount?: number;
  isHighlight: boolean;
  positionId?: number;
}

export interface CandidateJob {
  id: number;
  jobId: number;
  cvId?: number;
  status: CandidateJobStatus;
  appliedAt: string;
  updatedAt: string;
}

export interface SavedJob {
  id: number;
  jobId: number;
}

File: JobMatchingSystem.API/Models/Job.cs
using JobMatchingSystem.API.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JobMatchingSystem.API.Models
{
    public class Job
    {
        [Key]
        public int JobId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Requirements { get; set; }
        public string Benefits { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public int? SalaryMin { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public int? SalaryMax { get; set; }
        public string Location { get; set; }
        public int? ExperienceYear { get; set; }
        public string JobType { get; set; }
        public JobStatus Status { get; set; } = JobStatus.Draft;
        public int? PositionId { get; set; }
        public int ViewsCount { get; set; } = 0;
        public int CompanyId { get; set; }
        public int RecuiterId { get; set; }
        public int? VerifiedBy { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? OpenedAt { get; set; }
        public DateTime? ExpiredAt { get; set; }

        public bool IsHighlighted { get; set; } = false;
        // Thời điểm Job nổi bật sẽ kết thúc
        public DateTime? HighlightedUntil { get; set; }

        public bool IsDeleted { get; set; } = false;

        // Navigation properties
        [ForeignKey("CompanyId")]
        public virtual Company? Company { get; set; } = null!;
        [ForeignKey("RecuiterId")]
        public virtual ApplicationUser? Recruiter { get; set; }
        [ForeignKey("VerifiedBy")]
        public virtual ApplicationUser? Admin { get; set; }
        public virtual ICollection<CandidateJob> CandidateJobs { get; set; } = new List<CandidateJob>();
        public virtual ICollection<SavedJob> SavedJobs { get; set; } = new List<SavedJob>();
        public virtual ICollection<Report> Reports { get; set; } = new List<Report>();
        public virtual ICollection<JobStage> JobStages { get; set; } = new List<JobStage>();
        public virtual ICollection<JobTaxonomy> JobTaxonomies { get; set; } = new List<JobTaxonomy>();

        [ForeignKey("PositionId")]
        public virtual Position? Position { get; set; }
    }
}